### ISSUE TITLE
Downloading says "finished" but percent is "0" and file does not appear.

### ISSUE THREAD

**Issue by baobabKoodaa** [MrS0m30n3/youtube-dl-gui#419]
I just did a fresh reinstall of youtube-dl-gui on Windows using the Windows installer. When I try to download a youtube video, the status says "Finished" while Percent says "0%" and the file doesn't appear. I'm guessing that youtube-dl is throwing some kind of error, which is supressed by youtube-dl-gui. I recommend that you don't supress errors like this, so that the user has some chance to fix them.

**Comment by jk-85** [comment]
Me too has a problem with this one: https://youtu.be/D2KRO0qRDhU

**Comment by PacoH** [comment]
Yup. Me too. Says "Installation Windows 	run setup.exe." Nope. Installed vcredist_x86.exe, ffmpeg.exe. Still same. Tried portable. Still same. Does this work for anyone else?

**Comment by TheLichKingZeyd** [comment]
wierd i didn't get this issue?
what OS was it?
did you install the latest updates of what this software need, i think you need Visual basic or python if you used the none windows installer

**Comment by PacoH** [comment]
As I said, Windows (10) with Visual C++ Redistributable Package vcredist_x86. I added the installed directory to the path. Just running the installer is supposed to work.

**Comment by Lifell** [comment]
Did you click the little cog wheel and then "Update"?

I feel that most of the "issues" here are just that. Outdated youtube-dl. I tried: https://youtu.be/D2KRO0qRDhU and it worked fine.

**Comment by baobabKoodaa** [comment]
> Did you click the little cog wheel and then "Update"?
> 
> I feel that most of the "issues" here are just that. Outdated youtube-dl. I tried: https://youtu.be/D2KRO0qRDhU and it worked fine.

That's unnecessarily hostile and inaccurate. I was very clear about the issue: _"I recommend that you don't supress errors like this, so that the user has some chance to fix them."_ In other words, I reported an issue that youtube-dl-gui is supressing errors instead of showing them to the user. Clicking the little cog wheel is not going to stop youtube-dl-gui from supressing warnings and errors.

**Comment by axcore** [comment]
Guys guys. youtube-dl-gui hasn't been updated for years. There is a fork [here](https://github.com/axcore/tartube), maybe you can try that.

**Comment by PacoH** [comment]
OK, it works but [Youtube-dl-gui](https://dev.to/aswathm78/youtube-dl-gui-a-sane-way-to-download-youtube-videos-in-bulk-3cja) is dated Jul 20 '18. Usually if there is a newer fork, it is linked to so people like us don't have to spend hours trying to get the old fork to work. Please update that.

**Comment by axcore** [comment]
I don't think that's possible after the fact.

**Comment by PacoH** [comment]
Did you click the little cog wheel and then "Update"?

I tried that with but still didn't work.

**Comment by PacoH** [comment]
"I don't think that's possible after the fact."

What do you mean "after the fact"?

**Comment by resynth1943** [comment]
Yeah, this has been happening on multiple Linux-based systems too, which all (to my knowledge) are using v0.4. I cannot download *anything*, no matter what the options are. What the hell has changed here? This used to work well. Sad.

I'm starting to suspect YouTube have had a hand in this. I see errors in the logs about not being able to download due to *copyright* issues (although the company referenced changes each time). Moreover, I haven't been able to use `youtube-dl` on the Mac. It simply says "This video is unavailable", although with specific options it seems to work fine. I'll look into this over the weekend and report back.

**Comment by baobabKoodaa** [comment]
If anyone is looking for an alternative, I decided to create [one-click-youtube-dl](https://github.com/baobabKoodaa/one-click-youtube-dl) after being frustrated with the GUI.

**Comment by PacoH** [comment]
>I haven't been able to use youtube-dl on the Mac. It simply says "This video is unavailable", although with specific options it seems to work fine.

I just downloaded the default quality video from a random video `youtube-dl https://www.youtube.com/watch?v=jjdWckwmhB0`.
	
	OS X 10.8.5
	youtube-dl version 2019.11.05
	Python version 2.7.14 (CPython) - Darwin-12.6.0-x86_64-i386-64bit
	exe versions: ffmpeg 3.4.1, ffprobe 3.4.1, phantomjs 2.1.1, rtmpdump 2.4

**Comment by axcore** [comment]
> What do you mean "after the fact"?

How can Tartube be marked as a fork of youtube-dl-gui, so that you would have noticed it?

**Comment by PacoH** [comment]
Usually there is a message that the tool is no longer updated with a link to a fork of it.

**Comment by axcore** [comment]
> Usually there is a message that the tool is no longer updated with a link to a fork of it.

Well, that would be great, but I don't have access to the youtube-dl-gui pages. @MrS0m30n3 can edit them, but I can't.

**Comment by PacoH** [comment]
I see [@MrS0m30n3](https://github.com/MrS0m30n3) has had no activity since July 2018. Is there a way to contact him?

**Comment by Lifell** [comment]
> 
> 
> > Did you click the little cog wheel and then "Update"?
> > I feel that most of the "issues" here are just that. Outdated youtube-dl. I tried: https://youtu.be/D2KRO0qRDhU and it worked fine.
> 
> That's unnecessarily hostile and inaccurate. I was very clear about the issue: _"I recommend that you don't supress errors like this, so that the user has some chance to fix them."_ In other words, I reported an issue that youtube-dl-gui is supressing errors instead of showing them to the user. Clicking the little cog wheel is not going to stop youtube-dl-gui from supressing warnings and errors.

I don't see how this was hostile. It read my reply again and I really don't see it. It wasn't meant as a hostile thing either. You're right regarding that the errors should be shown though but that wasn't what I commented on and that didn't seem to be the main problem here. It doesn't work. The title focuses on it not working and so does most of your original post.

The reason why I said this is because I think the "Update" portion of the GUI is badly implemented and I thought a lot of people would overlook that. So I thought I would mention that. I do see however that I worded it badly but it doesn't come across to me as hostile at all. I wanted to help =/.

Either way, I do think errors should be shown yeah but it's unfortunately doubtful that that will happen because the author of this GUI hasn't updated it in quite awhile. Which is a shame. But yeah, sorry if I seemed hostile to you. Wasn't meant that way. Giving the user (useful) feedback if errors occur is a good thing.

**Comment by baobabKoodaa** [comment]
Hey, no worries, and thanks for offering your help.

**Comment by PacoH** [comment]
I found a free app that replaces this and the recommended app–[WinX YouTube Downloader](https://www.winxdvd.com/youtube-downloader/). It is very professional looking and works with ease-[WinX YouTube Downloader Step\-by\-step User Guide](https://www.winxdvd.com/help/how-to-use-winx-youtube-downloader.htm). I was able to download 2.63 GB 1080p PBS shows instantly. It supports 300+ sites (a subset of the 1100+ youtube-dl Supported sites).

I was looking for a Windows app that made it easy for a non-technical person, my brother, to download videos. This is it. Highly recommended.

I tried the MacX version but it didn't work with PBS.

**Comment by resynth1943** [comment]
@PacOh I probably should've gone into more detail.

Although that appears to function in the expected manner, have you tried using the `--format=mp3` flag with that command? That has failed numerous times. As far as I can tell, that's the right way to do it. Currently, I'm downloading them as M4A files and converting them to MP3 files, so any assistance would be most appreciated.

Many thanks,
Resynth.

**Comment by PacoH** [comment]
@resynth1943

You have to read the man page. `/` to search. `n` for next item. Search [i]mp3[/i]. Youtube audio formats are never mp3 to my knowledge. You have to do post-processing using `ffmpeg`.

Try this:`youtube-dl --extract-audio --audio-format "mp3" --audio-quality 5 https://www.youtube.com/watch?v=hcgvYr2nlrk`. It works. You will get an `mp3` file only. You must have `ffmpeg` installed but you probably do already because it is necessary to combine audio and video.

**Comment by PacoH** [comment]
@resynth1943

You have to read the man page. `/` to search. `n` for next item. Search for *mp3*. Youtube audio formats are never `mp3` to my knowledge. You have to do post-processing using `ffmpeg`.

Try this:`youtube-dl --extract-audio --audio-format "mp3" --audio-quality 5 https://www.youtube.com/watch?v=hcgvYr2nlrk`. It works. You will get an `mp3` file only. You must have `ffmpeg` installed but you probably do already because it is necessary to combine audio and video.

**Comment by resynth1943** [comment]
@PacoH Thanks, dude! I'll be checking that out next time I need to download some music. :heart:

**Comment by tbategit** [comment]
Was having the same issue, I replaced the youtube-dl.exe file that this gui downloads automatically (7843kb), with the youtube-dl.exe version that TarTube downloads (101kb) in this folder on win10:

C:\Users\{YourWindowsUsername}\AppData\Roaming\youtube-dlg

Working fine now. 

So it seems that the server where the latest dl exe gets downloaded from 
(probably here: https://yt-dl.org/downloads/latest/youtube-dl.exe but I haven't looked at the sourcecode) 
has been compromised in some way.

**Comment by sergencetn** [comment]
In https://ytdl-org.github.io/youtube-dl/download.html ,

> Windows exe requires Microsoft Visual C++ 2010 Redistributable Package (x86) and does not require Python that is already embedded into the binary.

So you should install  Microsoft Visual C++ 2010 Redistributable Package (x86) from https://www.microsoft.com/en-US/download/details.aspx?id=5555 .
